worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -129,6 +131,35 @@ public PartitionFilesSorter(
         ThreadUtils.newDaemonCachedThreadPool(
             "worker-file-sorter-execute", conf.partitionSorterThreads(), 120);
 
+    indexCache =
+        CacheBuilder.newBuilder()
+            .concurrencyLevel(conf.partitionSorterThreads())
+            .expireAfterAccess(conf.partitionSorterIndexExpire(), TimeUnit.MILLISECONDS)
+            .maximumWeight(indexCacheMaxWeight)
+            .weigher(
+                (key, cache) -> {
+                  // estimated memory usage
+                  int weight = 0;
```

Comment:
It's very hard to estimate real memory usage inside JVM, I doubt whether this piece of code correctly calculates real memory usage. IMO @ErikFang 's suggestion is simpler and better, using number of `ShuffleBlockInfo` is enough and easy to tune the configuration.